### ISSUE TITLE
(one specific) Cover on spinny is squeezed

### ISSUE THREAD

**Issue by ronso0** [mixxxdj/mixxx#11967]
The cover is displayed correctly in
* deck cover art widget
* library cover art widget
* Track Properties dialog

just in the spinny it's sueezed vertically.
Never saw this before, all other covers look good.

I noticed the issue while RGB (GLSL) waveforms were selected, but selecting RGB (legacy, GL) or just RGB doesn't fix it.

This is how it looks like in Tango
![Screenshot_2023-09-11_22-29-48](https://github.com/mixxxdj/mixxx/assets/5934199/599dc9e5-ed0d-4f6f-80a5-f04a9ce6c5ce)

This is the cover, maybe someone can reproduce the issue:
[squeezed-spinny-cover.jpg.zip](https://github.com/mixxxdj/mixxx/files/12580261/squeezed-spinny-cover.jpg.zip)

**Comment by Swiftb0y** [comment]
Could it just be that cover is non-square and the spinny scales it instead of whatever the other widgets are doing (cropping?)?

**Comment by ronso0** [comment]
Well, the images is 745 x 640 and it's squeezed beyond recognisability (I added a screenshot just now :facepalm: ).
![Screenshot_2023-09-11_22-29-48](https://github.com/mixxxdj/mixxx/assets/5934199/5cc4e602-fd30-428e-9817-1a5896bfdc59)


Here's how it looks like when I crop it to width < height (545 x 640)
![image](https://github.com/mixxxdj/mixxx/assets/5934199/addf3827-7f5e-497d-8ea7-11e6129233be)

**Comment by Swiftb0y** [comment]
Does it work when you crop it to be rectangular?

**Comment by ronso0** [comment]
Yup.

**Comment by Swiftb0y** [comment]
can confirm/reproduce locally. doesn't seem terribly hard to fix, but I don't know where to look or where the erroneous scaling could be introduced. maybe @m0dB can give me some pointers?

**Comment by m0dB** [comment]
The bug must be here in WSpinnyGLSL::drawTexture.

```
    // fill centered
    const float posx2 = tw >= th ? 1.f : (th - tw) / th;
    const float posy2 = th >= tw ? 1.f : (tw - th) / tw;
    const float posx1 = -posx2;
    const float posy1 = -posy2;
```

The target coordinates of the widget viewport are -1,-1 to 1,1.

So for example an image of 100x200 should be drawn at -0.5,-1 0.5,1

**Comment by m0dB** [comment]
So yeah, this code is wrong. Not sure what I was thinking!

**Comment by m0dB** [comment]
So I believe the correct code simply is:
```
const float posx2 = tw >= th ? 1.f : tw / th;
const float posy2 = th >= tw ? 1.f : th / tw;
```

I don't have the time to try this now, maybe you can give it a go @Swiftb0y ?

**Comment by Swiftb0y** [comment]
Sure, I'll test later.

**Comment by Swiftb0y** [comment]
fixed by #11971
